src/test/java/ca/nexapp/core/persistence/unitofwork/rest/TransactionFilterTest.java

```diff
@@ -0,0 +1,94 @@
+package ca.nexapp.core.persistence.unitofwork.rest;
+
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.*;
+
+import java.io.IOException;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import ca.nexapp.core.persistence.unitofwork.UnitOfWork;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TransactionFilterTest {
+
+    @Mock
+    private UnitOfWork<Connection> unitOfWork;
+
+    private ServletRequest servletRequest;
+    private ServletResponse servletResponse;
+    private FilterChain filterChain;
```

Comment:
Why do you use `@Mock` for the first item, but not for the others here?

Reply:
limitation of the `mock(...)` methods and Java type-system in general. Easiest method to mock a templated class